src/helics/apps/helicsApp.cpp

```diff
@@ -31,185 +31,186 @@ std::map<std::string, int> pubids;
+        return;
     }
 
-    void App::processArgs(std::unique_ptr<helicsCLI11App>& app, const std::string& defaultAppName)
-    {
-        remArgs = app->remaining_for_passthrough();
-        auto ret = app->last_output;
-        if (ret == helicsCLI11App::parse_output::help_call) {
-            if (!app->quiet) {
-                // this is just to run the help output
-                FederateInfo helpTemp("--help");
-                (void)helpTemp;
-            }
-            helpMode = true;
-        }
-        if (ret != helicsCLI11App::parse_output::ok) {
-            deactivated = true;
-            return;
-        }
-
-        if (masterFileName.empty()) {
-            if (!fileLoaded) {
-                if (CLI::ExistingFile("helics.json").empty()) {
-                    masterFileName = "helics.json";
-                }
+    if (masterFileName.empty()) {
+        if (!fileLoaded) {
+            if (CLI::ExistingFile("helics.json").empty()) {
+                masterFileName = "helics.json";
             }
         }
-
-        FederateInfo fi(remArgs);
-        if (fi.defName.empty()) {
-            fi.defName = defaultAppName;
-        }
-
-        fed = std::make_shared<CombinationFederate>("", fi);
     }
 
-    App::App(const std::string& appName, const FederateInfo& fi):
-        fed(std::make_shared<CombinationFederate>(appName, fi))
-    {
+    if (fi.defName.empty()) {
+        fi.defName = defaultAppName;
     }
 
-    App::App(const std::string& appName, const std::shared_ptr<Core>& core, const FederateInfo& fi):
-        fed(std::make_shared<CombinationFederate>(appName, core, fi))
-    {
+    fed = std::make_shared<CombinationFederate>("", fi);
+}
+
+App::App(const std::string& appName, const FederateInfo& fi):
+    fed(std::make_shared<CombinationFederate>(appName, fi))
+{
+}
+
+App::App(const std::string& appName, const std::shared_ptr<Core>& core, const FederateInfo& fi):
+    fed(std::make_shared<CombinationFederate>(appName, core, fi))
+{
+}
+
+App::App(const std::string& appName, CoreApp& core, const FederateInfo& fi):
+    fed(std::make_shared<CombinationFederate>(appName, core, fi))
+{
+}
+
+App::App(const std::string& appName, const std::string& jsonString):
+    fed(std::make_shared<CombinationFederate>(appName, jsonString))
+{
+    if (jsonString.size() < 200) {
+        masterFileName = jsonString;
     }
-
-    App::App(const std::string& appName, CoreApp& core, const FederateInfo& fi):
-        fed(std::make_shared<CombinationFederate>(appName, core, fi))
-    {
+}
+
+App::~App() = default;
+
+std::unique_ptr<helicsCLI11App> App::generateParser()
+{
+    auto app =
+        std::make_unique<helicsCLI11App>("common options for all Helics Apps", "[HELICS_APP]");
+
+    app->add_flag("--local",
+                  useLocal,
+                  "specify otherwise unspecified endpoints and publications as local( "
+                  "i.e.the names will be prepended with the player name)");
```

Comment:
```suggestion
                  "Specify otherwise unspecified endpoints and publications as local "
                  "(i.e. the names will be prepended with the player name)");
```